jmxtrans-output/jmxtrans-output-core/src/main/java/com/googlecode/jmxtrans/model/output/support/TcpOutputWriterBuilder.java

```diff
@@ -70,11 +74,19 @@ private TcpOutputWriterBuilder(@Nonnull InetSocketAddress server, @Nonnull T tar
 						socketTimeoutMillis,
 						charset,
 						flushStrategy)))
-				.setExpiration(new SocketExpiration())
+				.setExpiration(createSocketExpiration())
 				.setSize(poolSize);
 		return new BlazePool<>(config);
 	}
 
+	private Expiration<SocketPoolable> createSocketExpiration() {
+		if (socketExpirationMs <= 0) {
+			return new SocketExpiration();
+		} else {
+			return new TimeExpiration<SocketPoolable>(socketExpirationMs, MILLISECONDS);
```

Comment:
We probably want a compound expiration that still does the checks in `SocketExpiration` in addition to the timeout.

Reply:
Right, changed it to use a `CompoundExpiration`.